pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -38,6 +38,8 @@ const (
 	UndefinedParameter BuildReason = "UndefinedParameter"
 	// RemoteRepositoryUnreachable indicates the referenced repository is unreachable
 	RemoteRepositoryUnreachable BuildReason = "RemoteRepositoryUnreachable"
+	// BuildNameInvalid indicates the build name is invalid
+	BuildNameInvalid BuildReason = "BuildNameInvalid"
```

Comment:
Please add documentation for the new error reason to the table here: https://github.com/shipwright-io/build/blob/main/docs/build.md#build-validations